pkgs/by-name/we/wechat/package.nix

```
@@ -49,8 +18,43 @@ appimageTools.wrapAppImage {
     maintainers = with lib.maintainers; [ prince213 ];
     mainProgram = "wechat";
     platforms = [
+      "aarch64-darwin"
+      "x86_64-darwin"
       "aarch64-linux"
       "x86_64-linux"
     ];
   };
+
+  sources =
+    let
+      any-darwin = {
+        version = "4.0.5.24";
+        src = fetchurl {
+          url = "https://web.archive.org/web/20250608064358if_/https://dldir1v6.qq.com/weixin/Universal/Mac/WeChatMac.dmg";
+          hash = "sha256-ieixBgYhZ5jU3TWCV7BXKFBidJ1bbabXBHTkrpNcGDI=";
+        };
+      };
+    in
+    {
+      aarch64-darwin = any-darwin;
+      x86_64-darwin = any-darwin;
+      aarch64-linux = {
+        version = "4.0.1.11";
+        src = fetchurl {
+          url = "https://web.archive.org/web/20250512112413if_/https://dldir1v6.qq.com/weixin/Universal/Linux/WeChatLinux_arm64.AppImage";
+          hash = "sha256-Rg+FWNgOPC02ILUskQqQmlz1qNb9AMdvLcRWv7NQhGk=";
+        };
+      };
+      x86_64-linux = {
+        version = "4.0.1.11";
+        src = fetchurl {
+          url = "https://web.archive.org/web/20250512110825if_/https://dldir1v6.qq.com/weixin/Universal/Linux/WeChatLinux_x86_64.AppImage";
+          hash = "sha256-gBWcNQ1o1AZfNsmu1Vi1Kilqv3YbR+wqOod4XYAeVKo=";
+        };
+      };
+    };
+in
+callPackage (if stdenvNoCC.hostPlatform.isDarwin then ./darwin.nix else ./linux.nix) {
+  inherit pname meta;
+  inherit (sources.${system}) version src;
```

```suggestion
  inherit (sources.${system} or throw "Unsupported system: ${system}") version src;
```

Here `sources.${system}` may break eval for exotic platforms, as here you don't have the `meta.platforms` guard from the derivation...